web-ui/src/components/admin/roles/Roles.jsx

```diff
@@ -118,28 +139,29 @@ const Roles = () => {
         type: UPDATE_TOAST,
         payload: {
           severity: "success",
-          toast: `${member.name} added to ${selectedRole}s`,
+          toast: `${member.name} added to ${currentRole.role}s`,
         },
       });
     }
     setSelectedMember({});
   };
 
-  const addRole = async (member, role) => {
-    let res = await addNewRole(selectedRole, member.id, csrf);
+  const saveRole = async (role) => {
+    let res = role.id ? await updateRole(role, csrf) : await addNewRole(role, csrf);
     let data =
       res.payload && res.payload.data && !res.error ? res.payload.data : null;
     if (data) {
-      setShowAddRole(false);
+      setShowEditRole(false);
+      const updatedRoles = [...roles.filter((role) => role?.id !== data.id), data];
       dispatch({
-        type: UPDATE_ROLES,
-        payload: data,
+        type: SET_ROLES,
+        payload: updatedRoles,
       });
       window.snackDispatch({
         type: UPDATE_TOAST,
         payload: {
           severity: "success",
-          toast: `${member.name} added to ${selectedRole}s`,
+          toast: `Role Updated: ${role.role}`,
```

Comment:
Do we want this toast to instead say "Role created"?

Reply:
Nope. It's used for updates and additions.